Update AlwaysOn documentation to move away from the listener endpoint (#30041)
DataDog/documentation: layouts/shortcodes/dbm-sqlserver-agent-config-examples.en.md

### Using AlwaysOn

> For AlwaysOn users, the Agent should be installed on each replica server and connected directly to each replica. The full set of AlwaysOn telemetry will be collected from each individual replica, in addition to host-based telemetry (CPU, disk, memory etc.) for each server.

Reviewer: ```suggestion
For AlwaysOn users, the Agent should be installed on each replica server and connected directly to each replica. The full set of AlwaysOn telemetry is collected from each individual replica, in addition to host-based telemetry (CPU, disk, memory, and so on) for each server.
```
